src/Aeson.purs

```diff
@@ -134,31 +159,47 @@ jsonToAeson = stringify >>> decodeJsonString >>> fromRight shouldNotHappen
   -- valid json should always decode without errors
   shouldNotHappen = undefined
 
+getNumberIndex :: Aeson -> NumberIndex
+getNumberIndex (Aeson { numberIndex }) = numberIndex
+
 -------- Aeson manipulation and field accessors --------
 
--- TODO: add getFieldOptional if ever needed.
 getField
   :: forall (a :: Type)
    . DecodeAeson a
   => Object Aeson
   -> String
   -> Either JsonDecodeError a
 getField aesonObject field = getField' decodeAeson aesonObject field
+  where
+  -- | Adapted from `Data.Argonaut.Decode.Decoders`
+  getField'
+    :: (Aeson -> Either JsonDecodeError a)
+    -> Object Aeson
+    -> String
+    -> Either JsonDecodeError a
+  getField' decoder obj str =
+    maybe
+      (Left $ AtKey str MissingValue)
+      (lmap (AtKey str) <<< decoder)
+      (FO.lookup str obj)
 
 infix 7 getField as .:
 
--- | Adapted from `Data.Argonaut.Decode.Decoders`
-getField'
-  :: forall (a :: Type)
-   . (Aeson -> Either JsonDecodeError a)
-  -> Object Aeson
-  -> String
-  -> Either JsonDecodeError a
-getField' decoder obj str =
-  maybe
-    (Left $ AtKey str MissingValue)
-    (lmap (AtKey str) <<< decoder)
-    (FO.lookup str obj)
+getFieldOptional :: forall a. DecodeAeson a => Object Aeson -> String -> Either JsonDecodeError (Maybe a)
```

Comment:
Stylistic: here and below there are a large number of type variables without kind signatures. I think it's worth adding them for consistency with the rest of the codebase